### PR TITLE
Ensure kit selection quickPick doesn't show on non CMake projects

### PR DESCRIPTION
**Pull request by andreeis**
Fix for https://github.com/microsoft/vscode-cmake-tools/issues/1578.
Plus the following:
- rename for commands related to edit cache UI, they were confusingly named openConfiguration
- edit cache was not guarded by full feature set, being available in the UI for non CMake projects
- resetState command is useful even if the current project is seen as non CMake (mostly to identify bugs when the extension analyzes wrong this aspect). It's inconvenient to have resetState not available (it's hard to get out of a bad state when investigating some bugs)
- the extension would show full features (until the next reload) even if the single project in the workspace would be disabled, because we were calculating the state of the workspace too early

### CODE REVIEW

File: src/cmake-tools.ts [hunk 1]
```diff
@@ -74,6 +74,7 @@ export enum ConfigureTrigger {
   commandCleanConfigure = "commandCleanConfigure",
   commandConfigureAll = "commandConfigureAll",
   commandCleanConfigureAll = "commandCleanConfigureAll",
+  kitSelection = "kitSelection",
```
Comment (bobbrow) [inline src/cmake-tools.ts:77]: This doesn't appear to be needed anymore

File: src/extension.ts [hunk 2]
```diff
@@ -304,7 +306,7 @@ class ExtensionManager implements vscode.Disposable {
       // We have an active kit. We're good.
       return true;
     }
-    // No kit? Ask the user what they want.
+    // No kit? Ask the user what they want, if showQuickPick allows.
```
Comment (bobbrow) [inline src/extension.ts:309]: I think this comment should be reverted now

File: src/extension.ts [hunk 3]
```diff
@@ -406,30 +416,56 @@ class ExtensionManager implements vscode.Disposable {
...
+      buildKitTargetArch: "",
+      buildKitVersionMajor: "",
+      buildKitVersionMinor: "",
+      workspaceHash: "",
+    };
+
+    // Don't configure if the current project is not CMake based (it doesn't have a CMakeLists.txt in the sourceDirectory).
+    const sourceDirectory: string = cmt.workspaceContext.config.sourceDirectory;
+    const expandedSourceDirectory: string = util.lightNormalizePath(await expandString(sourceDirectory, { vars: optsVars }));
+    if (await fs.exists(expandedSourceDirectory + "/CMakeLists.txt")) {
```
Comment (bobbrow) [inline src/extension.ts:445]: We also support sourceDirectory with `CMakeLists.txt` at the end (even though it's technically incorrect), so you should take that into account here as well.

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: src/extension.ts (hunk 3 above)
Comment (andreeis) [reply src/extension.ts:445]: Ok, can it ever be a different file name than "CMakeLists.txt"? I wonder if we should error for such a setting.

File: src/extension.ts [hunk 4]
```diff
@@ -441,8 +441,11 @@ class ExtensionManager implements vscode.Disposable {
 
     // Don't configure if the current project is not CMake based (it doesn't have a CMakeLists.txt in the sourceDirectory).
     const sourceDirectory: string = cmt.workspaceContext.config.sourceDirectory;
-    const expandedSourceDirectory: string = util.lightNormalizePath(await expandString(sourceDirectory, { vars: optsVars }));
-    if (await fs.exists(expandedSourceDirectory + "/CMakeLists.txt")) {
+    let expandedSourceDirectory: string = util.lightNormalizePath(await expandString(sourceDirectory, { vars: optsVars }));
+    if (path.basename(expandedSourceDirectory).toLocaleLowerCase() !== "cmakelists.txt") {
+      expandedSourceDirectory = path.join(expandedSourceDirectory, "/CMakeLists.txt");
```
Comment (bobbrow) [inline src/extension.ts:446]: `path.join` shouldn't need you to add the directory separator in `"/CMakeLists.txt"`

**Review by bobbrow — APPROVED** [review]

**Review by bobbrow — APPROVED** [review]

**Pull request merged**